go/private/context.bzl

```diff
@@ -827,6 +799,34 @@ def _cgo_context_data_impl(ctx):
         cc_toolchain.target_gnu_system_name,
     )
 
+    # Add C toolchain directories to PATH.
```

Comment:
I'm planning to get rid of `cgo_context` in favor of resolving the toolchain information for each individual target. Should I take this change to mean that this could have a relevant impact on analysis time?

Reply:
This change is measurable but not super impactful. Perhaps 50-100ms on buildbuddy repo. But if we do more toolchain resolution per target, it might have more impact. Why do you want to go that route?

Reply:
We can still merge this though as it improves the situation and doesn't make this future change more difficult.

Reply:
Ah ok, the desire makes sense. Let's profile with your change once it's ready and see how things look